bundles/org.openhab.binding.automower/src/main/java/org/openhab/binding/automower/internal/bridge/AutomowerBridgeHandler.java

```diff
@@ -0,0 +1,156 @@
+            }
+        }
+    }
+
+    public AutomowerBridgeHandler(Bridge bridge, HttpClient httpClient) {
+        super(bridge);
+        this.httpClient = httpClient;
+    }
+
+    @Override
+    public void dispose() {
+        if (bridge != null) {
+            stopAutomowerBridgePolling(bridge);
+            bridge = null;
+        }
+    }
+
+    @Override
+    public void initialize() {
+        bridgeConfiguration = getConfigAs(AutomowerBridgeConfiguration.class);
+
+        final String appKey = bridgeConfiguration.getAppKey();
+        final String userName = bridgeConfiguration.getUserName();
+        final String password = bridgeConfiguration.getPassword();
+        if (appKey == null || appKey.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "@text/conf-error-no-app-key");
+        } else if (userName == null || userName.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "@text/conf-error-no-username");
+        } else if (password == null || password.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "@text/conf-error-no-password");
+        } else {
+            if (bridge == null) {
+                bridge = new AutomowerBridge(appKey, userName, password, httpClient, scheduler);
+                startAutomowerBridgePolling(bridge);
+            }
+            updateStatus(ThingStatus.UNKNOWN);
+        }
+    }
+
+    private void startAutomowerBridgePolling(AutomowerBridge bridge) {
+        ScheduledFuture<?> currentPollingJob = automowerBridgePollingJob;
+        if (currentPollingJob == null) {
+            if (bridgeConfiguration.getPollingInterval() < 1) {
+                logger.info("No valid polling interval specified. Using default value: {}s", automowerPollingIntervalS);
+            } else {
+                automowerPollingIntervalS = bridgeConfiguration.getPollingInterval();
+            }
+            automowerBridgePollingJob = scheduler.scheduleWithFixedDelay(
+                    new AutomowerBridgePollingRunnable(this, bridge), 1, automowerPollingIntervalS, TimeUnit.SECONDS);
+        }
+    }
+
+    private void stopAutomowerBridgePolling(AutomowerBridge bridge) {
+        ScheduledFuture<?> currentPollingJob = automowerBridgePollingJob;
+        if (currentPollingJob != null) {
+            currentPollingJob.cancel(true);
+            automowerBridgePollingJob = null;
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    public AutomowerBridge getAutomowerBridge() {
+        return bridge;
+    }
+
+    public Optional<MowerListResult> getAutomowers() {
+        try {
+            return Optional.of(bridge.getAutomowers());
+        } catch (AutomowerCommunicationException e) {
+            logger.debug("Bridge cannot get list of available automowers", e);
```

Comment:
Do you log the stack trace by intention?

Reply:
no